pkg/cloudlogging/puller_test.go

```diff
@@ -126,12 +230,37 @@ func setupFakeClient(t *testing.T, ctx context.Context, s *fakeServer) *logging.
 type fakeServer struct {
 	loggingpb.UnimplementedLoggingServiceV2Server
 
-	req         *loggingpb.ListLogEntriesRequest
-	resp        *loggingpb.ListLogEntriesResponse
+	listReq     *loggingpb.ListLogEntriesRequest
+	listResp    *loggingpb.ListLogEntriesResponse
+	tailReq     *loggingpb.TailLogEntriesRequest
+	logCounter  int64
+	logNumCap   int64
```

Comment:
Since this is local, can we have a hardcoded cap? Like 10 should be enough

Reply:
const `fakeServerTailLogCap = 10`, and removed logNumCap from `fakeServer` field